core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/operators/aggregate/FinalAggregateOpExec.scala

```diff
@@ -5,21 +5,24 @@ import edu.uci.ics.amber.engine.common.InputExhausted
 import edu.uci.ics.amber.engine.common.rpc.AsyncRPCClient
 import edu.uci.ics.amber.engine.common.virtualidentity.LinkIdentity
 import edu.uci.ics.texera.workflow.common.operators.OperatorExecutor
-import edu.uci.ics.texera.workflow.common.operators.aggregate.PartialAggregateOpExec.INTERNAL_AGGREGATE_PARTIAL_OBJECT
+import edu.uci.ics.texera.workflow.common.operators.aggregate.PartialAggregateOpExec.constructInternalAggregatePartialObject
 import edu.uci.ics.texera.workflow.common.tuple.Tuple
 import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, Schema}
 
 import scala.collection.convert.ImplicitConversions.`collection AsScalaIterable`
 import scala.collection.{JavaConverters, mutable}
 
 class FinalAggregateOpExec[Partial <: AnyRef](
-    val aggFunc: DistributedAggregation[Partial]
+    val aggFuncs: List[DistributedAggregation[Partial]]
 ) extends OperatorExecutor {
 
   var groupByKeyAttributes: Array[Attribute] = _
   var schema: Schema = _
 
-  var partialObjectPerKey = new mutable.HashMap[List[AnyRef], Partial]()
+  // partialObjectsPerKey has the same length as aggFuncs
+  // partialObjectsPerKey[i] corresponds to aggFuncs[i]
+  var partialObjectsPerKey =
```

Comment:
add a type declaration for this varaible, it's hard to understand after adding `(1 to aggFuncs.length).map...`